File: nysol_python/mcmd/methods/convTex/db/smp/mpadding.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mpadding'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
no
3
6
8
'''
db['idatas'].append(data)

data={}
data['name']='dat2.csv'
data['text']='''
date,dummy
20130929,a
20131002,b
20131004,c
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``no`` 項目が整数(\%n)として連続するようにレコードをパディングする。
``1`` とverb|4|の間に ``2,3`` を、 ``4`` と ``2`` の間に ``3`` が挿入されている。
'''
script['sh_code']='''
mpadding f=no%n i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mpadding(f="no%n", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='開始値、終了値の指定'
script['comment']='''
行間のパディングだけでなく、先頭行/終端行の前後もパディングする。
前後の範囲は ``S=,E=`` で指定する。
'''
script['sh_code']='''
mpadding f=no%n S=1 E=10 i=dat1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mpadding(f="no%n", S="1", E="10", i="dat1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='日付パディング'
script['comment']='''
``date`` 項目が日付(\%d)として連続するようにレコードをパディングする。
``k=,f=`` で指定した以外の項目は、直前の行の項目値でパディングする。
'''
script['sh_code']='''
mpadding f=date%d i=dat2.csv o=rsl3.csv
'''
script['py_code']='''
nm.mpadding(f="date%d", i="dat2.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

script={}
script['title']='パディング用文字列指定'
script['comment']='''
``v=`` にてパディング文字列を指定することもできる。
'''
script['sh_code']='''
mpadding f=date%d v=padding i=dat2.csv o=rsl4.csv
'''
script['py_code']='''
nm.mpadding(f="date%d", v="padding", i="dat2.csv", o="rsl4.csv").run()
'''
script['odatas']='rsl4.csv'
db['scripts'].append(script)

script={}
script['title']='パディングにNULL値を指定'
script['comment']='''
``n=True`` を指定してNULL値でパディングすることも可能。
'''
script['sh_code']='''
mpadding f=date%d -n i=dat2.csv o=rsl5.csv
'''
script['py_code']='''
nm.mpadding(f="date%d", n=True, i="dat2.csv", o="rsl5.csv").run()
'''
script['odatas']='rsl5.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mpadding.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
no
3
6
8
EOF
)}

File.open("dat2.csv","w"){|fpw| fpw.write(
<<'EOF'
date,dummy
20130929,a
20131002,b
20131004,c
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Create padding with integer values (\verb|type=int|) between records in \verb|no| column.
Insert \verb|4,5| between \verb|3| and \verb|6|, and \verb|7| between \verb|6| and \verb|8|.
EOF
scp=<<'EOF'
more dat1.csv
mpadding f=no%n i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Specify the starting and ending value"
comment=<<'EOF'
Insert padding between records as well as before and after the first and last records from the input data.
Specify the starting and ending range at \verb|S=,E=|.
EOF
scp=<<'EOF'
mpadding f=no%n S=1 E=10 i=dat1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)
############## Example 3
title="Padding with date"
comment=<<'EOF'
Create padding to fill in values between dates (\verb|type=date|) in the \verb|date| column.
Create padding values in columns other than those specified at \verb|k=,f=|.
EOF
scp=<<'EOF'
more dat2.csv
mpadding f=date%d i=dat2.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)

############## Example 4
title="Specify character string for padding"
comment=<<'EOF'
Specify the character string padding value at \verb|v=|.
EOF
scp=<<'EOF'
mpadding f=date%d v=padding i=dat2.csv o=rsl4.csv
more rsl4.csv
EOF
run(scp,title,comment)

############## Example 5
title="Specify NULL value as padding character"
comment=<<'EOF'
NULL value can be used as padding when the \verb|-n| option is specified.
EOF
scp=<<'EOF'
mpadding f=date%d -n i=dat2.csv o=rsl5.csv
more rsl5.csv
EOF
run(scp,title,comment)

